scripts/main.js

```diff
@@ -0,0 +1,348 @@
+          board[y][x] = 0; // Clear the original position
+          hasChanged = true;
+
+          // Animate the move without merging
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-up--${moveDistance}`); // Add a class for the move animation
+          }
+        }
+      }
+    }
+  }
+
+  return hasChanged; // Return whether the board changed
+}
+
+function moveDown() {
+  const originalBoard = board.map((row) => row.slice());
+  let hasChanged = false;
+
+  for (let x = 0; x < 4; x++) {
+    let merged = [false, false, false, false];
+
+    for (let y = 2; y >= 0; y--) {
+      if (board[y][x] !== 0) {
+        let newY = y;
+        let moveDistance = 0;
+
+        while (newY < 3 && board[newY + 1][x] === 0) {
+          newY++;
+          moveDistance++;
+        }
+
+        if (
+          newY < 3 &&
+          board[newY + 1][x] === board[y][x] &&
+          !merged[newY + 1]
+        ) {
+          board[newY + 1][x] *= 2;
+          scoreVal += board[newY + 1][x];
+          board[y][x] = 0;
+          merged[newY + 1] = true;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-down--${moveDistance + 1}`);
+          }
+        } else if (newY !== y) {
+          board[newY][x] = board[y][x];
+          board[y][x] = 0;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-down--${moveDistance}`);
+          }
+        }
+      }
+    }
+  }
+
+  return hasChanged;
+}
+
+function moveLeft() {
+  const originalBoard = board.map((row) => row.slice());
+  let hasChanged = false;
+
+  for (let y = 0; y < 4; y++) {
+    let merged = [false, false, false, false];
+
+    for (let x = 1; x < 4; x++) {
+      if (board[y][x] !== 0) {
+        let newX = x;
+        let moveDistance = 0;
+
+        while (newX > 0 && board[y][newX - 1] === 0) {
+          newX--;
+          moveDistance++;
+        }
+
+        if (
+          newX > 0 &&
+          board[y][newX - 1] === board[y][x] &&
+          !merged[newX - 1]
+        ) {
+          board[y][newX - 1] *= 2;
+          scoreVal += board[y][newX - 1];
+          board[y][x] = 0;
+          merged[newX - 1] = true;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-left--${moveDistance + 1}`);
+          }
+        } else if (newX !== x) {
+          board[y][newX] = board[y][x];
+          board[y][x] = 0;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-left--${moveDistance}`);
+          }
+        }
+      }
+    }
+  }
+
+  return hasChanged;
```

Comment:
The 'moveLeft' function, like the other move functions, has repeated logic for merging and moving cells. Abstracting common functionality into shared helper functions would make the code DRYer and easier to maintain.